CRM/Contact/BAO/Contact/Location.php

```diff
@@ -40,16 +40,22 @@ class CRM_Contact_BAO_Contact_Location {
    *
    * @param bool $isPrimary
    * @param int $locationTypeID
+   * @param bool $isBilling
    *
    * @return array
    *   Array of display_name, email, location type and location id if found, or (null,null,null, null)
    */
-  public static function getEmailDetails($id, $isPrimary = TRUE, $locationTypeID = NULL) {
+  public static function getEmailDetails($id, $isPrimary = TRUE, $locationTypeID = NULL, $isBilling = FALSE) {
     $primaryClause = NULL;
     if ($isPrimary) {
       $primaryClause = " AND civicrm_email.is_primary = 1";
     }
 
+    // CRM-20362 retrieving Billing email address.
+    if ($isBilling) {
+      $primaryClause = " AND civicrm_email.is_billing = 1";
```

Comment:
@eileenmcnaughton thanks for reviewing. 

> At the philosophical level - is_billing seems to have been a concept conceived but never really implemented (as summed up in "As that renders the billing email effectively useless.").

Agreed. Which greatly adds to the confusion of new CiviCRM users - why can I define billing email if it is not actually used at all by the system?  It does not make sense.

>  ie. if someone makes an online donation and they have a home address marked as is_billing is that updated or is a new one created, which steals the is_billing flag.

Even with this change, it is not possible to have a home email address marked as billing. They would be two separate email addresses.

> will there be unforseen circumstances if we start implementing it.

If a CiviCRM Contact has no billing email address, then there is no impact as it falls back to primary email as per current functionality.

If a CiviCRM Contact does have a billing email address set, then yes the contribution receipt would be sent to that address. Would that really be unexpected?

Since you can have an email address set as both the billing and primary, I dont' think this is major issue.